src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -309,6 +310,32 @@ EnsureModificationsCanRunOnRelation(Oid relationId)
 {
 	EnsureModificationsCanRun();
 
+	if (!OidIsValid(relationId) || !IsCitusTable(relationId))
+	{
+		/* we are not interested in PG tables */
+		return;
+	}
+
+	bool modifiedTableReplicated =
+		IsCitusTableType(relationId, REFERENCE_TABLE) ||
+		!SingleReplicatedTable(relationId);
+
+	if (!IsCoordinator() && !AllowModificationsFromWorkersToReplicatedTables &&
+		modifiedTableReplicated)
+	{
+		ereport(ERROR, (errmsg("modifications via the worker nodes are not "
```

Comment:
Well, we have this test in `local_shard_execution_replicated`. Not sure why the code-cov does't find it. It is in the mx schedule, could that be relevant to that?